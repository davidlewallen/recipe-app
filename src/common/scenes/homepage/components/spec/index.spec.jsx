import React from 'react';
import { shallow } from 'enzyme';

import Homepage from '../';

describe('Homepage snapshot test', () => {
  const props = {
    recipeURL: 'www.testrecipeurl.com',
    handleRecipe: () => {},
    submitRecipe: () => {},
    recipeList: [{ title: 'Recipe 1', _id: 1 }, { title: 'Recipe 2', _id: 2 }],
  };

  it('should match snapshot', () => {
    const component = shallow(<Homepage {...props} />);

    expect(component).toMatchSnapshot();
  });
});
