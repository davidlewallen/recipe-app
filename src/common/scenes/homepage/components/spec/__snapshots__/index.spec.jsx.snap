// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage snapshot test should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Homepage
    handleRecipe={[Function]}
    recipeList={
      Array [
        Object {
          "_id": 1,
          "title": "Recipe 1",
        },
        Object {
          "_id": 2,
          "title": "Recipe 2",
        },
      ]
    }
    recipeURL="www.testrecipeurl.com"
    submitRecipe={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <form
          className="submit-recipe"
        >
          <input
            onChange={[Function]}
            placeholder="Recipe URL"
            value="www.testrecipeurl.com"
          />
          <button
            onClick={[Function]}
          >
            Submit
          </button>
        </form>,
        <br />,
        <ul>
          <div>
            <li>
              Recipe 1
            </li>
            <button
              onClick={[Function]}
            >
              Delete
            </button>
          </div>
          <div>
            <li>
              Recipe 2
            </li>
            <button
              onClick={[Function]}
            >
              Delete
            </button>
          </div>
        </ul>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              onChange={[Function]}
              placeholder="Recipe URL"
              value="www.testrecipeurl.com"
            />,
            <button
              onClick={[Function]}
            >
              Submit
            </button>,
          ],
          "className": "submit-recipe",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "onChange": [Function],
              "placeholder": "Recipe URL",
              "value": "www.testrecipeurl.com",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Submit",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Submit",
            "type": "button",
          },
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              <li>
                Recipe 1
              </li>
              <button
                onClick={[Function]}
              >
                Delete
              </button>
            </div>,
            <div>
              <li>
                Recipe 2
              </li>
              <button
                onClick={[Function]}
              >
                Delete
              </button>
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li>
                  Recipe 1
                </li>,
                <button
                  onClick={[Function]}
                >
                  Delete
                </button>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Recipe 1",
                },
                "ref": null,
                "rendered": "Recipe 1",
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Delete",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Delete",
                "type": "button",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li>
                  Recipe 2
                </li>,
                <button
                  onClick={[Function]}
                >
                  Delete
                </button>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Recipe 2",
                },
                "ref": null,
                "rendered": "Recipe 2",
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Delete",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Delete",
                "type": "button",
              },
            ],
            "type": "div",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <form
            className="submit-recipe"
          >
            <input
              onChange={[Function]}
              placeholder="Recipe URL"
              value="www.testrecipeurl.com"
            />
            <button
              onClick={[Function]}
            >
              Submit
            </button>
          </form>,
          <br />,
          <ul>
            <div>
              <li>
                Recipe 1
              </li>
              <button
                onClick={[Function]}
              >
                Delete
              </button>
            </div>
            <div>
              <li>
                Recipe 2
              </li>
              <button
                onClick={[Function]}
              >
                Delete
              </button>
            </div>
          </ul>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                onChange={[Function]}
                placeholder="Recipe URL"
                value="www.testrecipeurl.com"
              />,
              <button
                onClick={[Function]}
              >
                Submit
              </button>,
            ],
            "className": "submit-recipe",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "onChange": [Function],
                "placeholder": "Recipe URL",
                "value": "www.testrecipeurl.com",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Submit",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Submit",
              "type": "button",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                <li>
                  Recipe 1
                </li>
                <button
                  onClick={[Function]}
                >
                  Delete
                </button>
              </div>,
              <div>
                <li>
                  Recipe 2
                </li>
                <button
                  onClick={[Function]}
                >
                  Delete
                </button>
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <li>
                    Recipe 1
                  </li>,
                  <button
                    onClick={[Function]}
                  >
                    Delete
                  </button>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Recipe 1",
                  },
                  "ref": null,
                  "rendered": "Recipe 1",
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Delete",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Delete",
                  "type": "button",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <li>
                    Recipe 2
                  </li>,
                  <button
                    onClick={[Function]}
                  >
                    Delete
                  </button>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Recipe 2",
                  },
                  "ref": null,
                  "rendered": "Recipe 2",
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Delete",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Delete",
                  "type": "button",
                },
              ],
              "type": "div",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
